web-vite/src/app/map/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { geojson as flatgeobuf } from "flatgeobuf";
+
+export function fbgBbox(map: any) {
+  const { lng, lat } = map.getCenter();
+  const { _sw, _ne } = map.getBounds();
+  const distanceX =
+    Math.min(Math.abs(_sw.lng - lng), Math.abs(_ne.lng - lng)) * 0.9;
+  const distanceY =
+    Math.min(Math.abs(_sw.lat - lat), Math.abs(_ne.lat - lat)) * 0.9;
+  return {
+    minX: lng - distanceX,
+    minY: lat - distanceY,
+    maxX: lng + distanceX,
+    maxY: lat + distanceY,
+  };
+}
+
+export async function getFgbData(map: any) {
+  let i = 0;
+  const fc = { type: "FeatureCollection", features: [] };
+
+  let iter = flatgeobuf.deserialize(
+    "https://storage.googleapis.com/osm-tardis/2013-02-03T15%3A00.fgb",
+    fbgBbox(map)
+  );
+
+  for await (let feature of iter) {
+    if (
+      feature.properties.type !== "relation" &&
+      (feature.properties.changeType === "added" ||
+        feature.properties.changeType === "modifiedNew" ||
+        feature.properties.changeType === "deletedNew")
+    ) {
+      fc.features = fc.features.concat({ ...feature, id: i });
```

Comment:
Curious about this... is there a reason you're using `concat` and re-assigning the array instead of 
`fc.features.push({ ...feature, id: i })`?

A quick benchmark makes it look like the `concat` approach is `O(n^2)` instead of `O(n)` for the `push`.

```js
function timeConcat(n) {
  console.time('timeConcat');
  let arr = [];
  for (let i = 0; i < n; i++) {
    arr = arr.concat([i]);
  }
  console.timeEnd('timeConcat');
}

function timePush(n) {
  console.time('timePush');
  let arr = [];
  for (let i = 0; i < n; i++) {
    arr.push(i);
  }
  console.timeEnd('timePush');
}
```

```
timeConcat(100000)
// timeConcat: 3213.114013671875 ms

timePush(100000)
// timePush: 2.22802734375 ms
```

Reply:
@kylebarron good catch. I generally prefer to use concat to avoid mutating the array, but in this case, performance is more important. I reverted to `.push()` in https://github.com/developmentseed/osm-gradient/pull/35.